File: introduction/ITP1_2_D-Circle in a Rectangle.cs
using System;

public class Program {
  public static void Main() {
    var line = Console.ReadLine().Split();

    int w = int.Parse(line[0]);
    int h = int.Parse(line[1]);
    int x = int.Parse(line[2]);
    int y = int.Parse(line[3]);
    int r = int.Parse(line[4]);

    if (r <= x && x <= w - r && r <= y && y <= h - r) {
      Console.WriteLine("Yes");
    } else {
      Console.WriteLine("No");
    }
  }
}

File: introduction/ITP1_2_D-Circle in a Rectangle.kt
fun main(args:Array<String>) {
  val line = readLine()!!.split(" ").map{it.toInt()}.toTypedArray()
  val w = line[0]
  val h = line[1]
  val x = line[2]
  val y = line[3]
  val r = line[4]
  if (r <= x && x <= w - r && r <= y && y <= h - r) {
    println("Yes")
  } else {
    println("No")
  }
}
